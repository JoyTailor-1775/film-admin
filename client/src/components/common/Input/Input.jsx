import React from 'react';
import './Input.scss';

/*
  A Input component needs next required parameters to be used:
    1) title: String - a title of the button.
    2) onChange : Function - a method, which will be called, when the input value is
        changed;
    3) value: any - a value of the input.
    4) name: - corredponds with <input/> html-attribute - 'name'.

  The Input also has optional parameters:
    1) type: String (default value - 'text') - parameter corresponds with 
        the html-attribute of the <input/> tag. The parameter may have next values:
        - number
        - text
    2) maxLength: String - parameter corresponds with the html-attribute of 
        the <input/> tag. 
    3) min , max : String - parameters are used with 'number' type, correspond
        with the html-attribute of the <input/> tag.
*/

const SUPPORTED_TYPES = Object.freeze({
  text: 'text',
  number: 'number',
});

const Input = ({ title, name, type, onChange, value, maxLength, min, max }) => {
  return (
    <div className="field">
      <label htmlFor={name} className="field__lable">
        {title}
      </label>
      <input
        type={SUPPORTED_TYPES[type] ? SUPPORTED_TYPES[type] : 'text'}
        name={name}
        onChange={onChange}
        value={value}
        className="field__input"
        maxLength={maxLength && maxLength}
        min={min && min}
        max={max && max}
      />
    </div>
  );
};

export default Input;
