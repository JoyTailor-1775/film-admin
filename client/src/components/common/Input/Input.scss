@import '../../../globals/variables.scss';

.field {
  padding: 5px 10px;
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  align-items: center;

  &__lable {
    font-weight: 700;
    color: $dark-red;
  }

  &__input {
    padding: 5px;
    box-sizing: border-box;
    border: solid 1px $dark-red;
    border-radius: 5px;
    width: 170px;

    &:focus {
      outline: none;
    }
  }
}
