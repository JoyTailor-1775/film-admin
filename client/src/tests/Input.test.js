import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../components/common/Input';

describe('<Input/>', () => {
  test('renders simple input', () => {
    const props = {
      title: 'Test Input',
      name: 'testInput',
      onChange: jest.fn(),
      value: 'testValue',
      maxLength: '20',
    };

    const component = renderer.create(<Input {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders number input', () => {
    const props = {
      title: 'Test Number Input',
      name: 'testInput',
      type: 'number',
      onChange: jest.fn(),
      value: 'testValue',
      min: '5',
      max: '20',
    };

    const component = renderer.create(<Input {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders invalid input', () => {
    const props = {
      title: 'Test Input',
      name: 'testInput',
      type: '232dfsdf',
      onChange: jest.fn(),
      value: 'testValue',
    };

    const component = renderer.create(<Input {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
