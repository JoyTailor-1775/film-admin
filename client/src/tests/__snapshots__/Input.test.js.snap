// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> renders invalid input 1`] = `
<div
  className="field"
>
  <label
    className="field__lable"
    htmlFor="testInput"
  >
    Test Input
  </label>
  <input
    className="field__input"
    name="testInput"
    onChange={[MockFunction]}
    type="text"
    value="testValue"
  />
</div>
`;

exports[`<Input/> renders number input 1`] = `
<div
  className="field"
>
  <label
    className="field__lable"
    htmlFor="testInput"
  >
    Test Number Input
  </label>
  <input
    className="field__input"
    max="20"
    min="5"
    name="testInput"
    onChange={[MockFunction]}
    type="number"
    value="testValue"
  />
</div>
`;

exports[`<Input/> renders simple input 1`] = `
<div
  className="field"
>
  <label
    className="field__lable"
    htmlFor="testInput"
  >
    Test Input
  </label>
  <input
    className="field__input"
    maxLength="20"
    name="testInput"
    onChange={[MockFunction]}
    type="text"
    value="testValue"
  />
</div>
`;
